pkgs/top-level/all-packages.nix

```diff
@@ -31595,40 +31595,56 @@ with pkgs;
   firefox-esr-102 = wrapFirefox firefox-esr-102-unwrapped { };
   firefox-esr-115 = wrapFirefox firefox-esr-115-unwrapped { };
 
-  firefox-bin-unwrapped = callPackage ../applications/networking/browsers/firefox-bin {
-    inherit (gnome) adwaita-icon-theme;
+  firefox-bin-unwrapped = callPackage ../applications/networking/browsers/firefox-bin/linux.nix {
     channel = "release";
     generated = import ../applications/networking/browsers/firefox-bin/release_sources.nix;
+    inherit (gnome) adwaita-icon-theme;
   };
 
-  firefox-bin = wrapFirefox firefox-bin-unwrapped {
-    pname = "firefox-bin";
-  };
+  firefox-bin = if stdenv.isDarwin then
+    callPackage ../applications/networking/browsers/firefox-bin/darwin.nix {
+      channel = "release";
+      generated = import ../applications/networking/browsers/firefox-bin/release_sources.nix;
+    }
+  else
+    wrapFirefox firefox-bin-unwrapped {
+      pname = "firefox-bin";
+    };
```

Comment:
Hm, this kinda violates the DRY principle. Perhaps turning wrapFirefox into the identity function on Darwin would be simpler?